ckan/logic/action/create.py

```diff
@@ -1000,7 +1001,17 @@ def user_create(context: Context,
     upload.upload(uploader.get_max_image_size())
 
     if not context.get('defer_commit'):
-        model.repo.commit()
+        try:
+            model.repo.commit()
+        except exc.IntegrityError as e:
```

Comment:
Can you extract this check into some small function, so that instead of these scary blocks `user_create` and `user_update` will do something like:

```
with logic.guard_agains_duplicated_email(data_dict["email"]):
    model.repo.commit()
```